diff --git a/node_modules/mindee/src/http/apiCore.js b/node_modules/mindee/src/http/apiCore.js
index be77aca..43240a3 100644
--- a/node_modules/mindee/src/http/apiCore.js
+++ b/node_modules/mindee/src/http/apiCore.js
@@ -21,15 +21,15 @@ export async function cutDocPages(inputDoc, pageOptions) {
  */
 export async function sendRequestAndReadResponse(dispatcher, options, url) {
     url ?? (url = `https://${options.hostname}${options.path}`);
-    const response = await request(url, {
+    const fetchOptions = {
         method: options.method,
         headers: options.headers,
-        headersTimeout: options.timeoutSecs * 1000,
         body: options.body,
-        dispatcher: dispatcher,
-    });
+    };
+    const response = await fetch(url, fetchOptions);
+    const responseMessageObj = { statusCode: response.status, headers: response.headers };
     logger.debug("Parsing the response ...");
-    let responseBody = await response.body.text();
+    let responseBody = await response.text();
     // handle empty responses from server, for example, in the case of redirects
     if (!responseBody) {
         responseBody = "{}";
@@ -38,14 +38,14 @@ export async function sendRequestAndReadResponse(dispatcher, options, url) {
         const parsedResponse = JSON.parse(responseBody);
         logger.debug("JSON parsed successfully, returning plain object.");
         return {
-            messageObj: response,
+            messageObj: responseMessageObj,
             data: parsedResponse,
         };
     }
     catch {
         logger.error("Could not parse the return as JSON.");
         return {
-            messageObj: response,
+            messageObj: responseMessageObj,
             data: { reconstructedResponse: responseBody },
         };
     }
